src/rust/runtime/memory/buffer/demibuffer.rs

```diff
@@ -724,7 +724,7 @@ impl Drop for DemiBuffer {
         match self.get_tag() {
             Tag::Heap => {
                 // This might be a chain of buffers.  If so, we'll walk the list.
-                let mut next_entry: Option<NonNull<MetaData>> = Some(self.ptr);
+                let mut next_entry: Option<NonNull<MetaData>> = Some(self.get_ptr());
```

Comment:
It appears that "ptr" is not supposed to be used directly (only via the get_tag()/get_ptr() functions). It would be good to make it difficult to use it directly (or at least obvious by having a indicative name like **ptr_with_tag or tagged_ptr** etc.).

Reply:
I was following the lead of the Rust standard library with the naming here.  Several things, such as Rc, Arc, etc., that are implemented using NonNull<Whatever> call that field "ptr".  But then, their pointers aren't tagged.

I think it would be reasonable to rename it tagged_ptr.  Might have even prevented this bug.  I'll do it tomorrow.

Reply:
I renamed it to tagged_ptr.